### DISCUSSION TITLE
Snowflake connector installed but not accessible

### DISCUSSION THREAD

**Discussion by paddelia** [apache/airflow#25491] [Q&A]
Hello, I have installed the snowflake connector and then I try to create a connection but Snowflake is not an option in the Admin/Connection/ pull down list. The whole system is new, running on Win10 with Docker/wsl2.
I do not know where to start troubleshooting. Any pointer? thanks

**Comment by Taragolis** [comment]
@paddelia did you install [apache-airflow-providers-snowflake](https://airflow.apache.org/docs/apache-airflow-providers-snowflake/stable/index.html) or just `snowflake-connector-python` ?

**Reply by Taragolis** [reply]
And also since you use docker you could check is it packages actually installed by run
```shell
docker exec -ti <your webserver container id or name> python -m pip list
```

If you use official Airflow images make sure that you've install packages by correct user `airflow` and not as `root`
https://airflow.apache.org/docs/docker-stack/build.html

**Reply by paddelia** [reply]
I do not see the connector in the list returned by docker exec -ti <your webserver container id or name> python -m pip list.
When I run the install for the modules I see that all prerequisite are already met.
I tried to force it by adding this to the compose, it did not work.
_PIP_ADDITIONAL_REQUIREMENTS: ${_PIP_ADDITIONAL_REQUIREMENTS:-apache-airflow-providers-snowflake -snowflake-sqlalchemy -snowflake-connector-python}

**Reply by potiuk** [reply]
Don't use _PIP_ADDITIONAL_REQUIREMENTS for anything else by a quick checks. Build your own image and use it if you want to use Docker Compose.

**Reply by potiuk** [reply]
There is a big fat warning printed if you try to do it, so PLEASE. JUST. DON'T. This is is a terrible antipattern.

**Reply by potiuk** [reply]
And look at the warnings in general in your logs when you run them. They will tell you most likely what the problem is you have.

**Reply by paddelia** [reply]
I set the compose back to '', reinstalled the snowflake connector and now it works.
I tried many things and I am a beginner so I am not 100% sure what did fix the issue.
I will try again with another connector.
Thank you for your help.

**Reply by potiuk** [reply]
It's the _PIP_ADDITIONAL_REQUIREMENTS is what makes it wrong. This is precisely the kind of things that you are going to get if you continue using it.
